dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/ContentletDetail.java

```diff
@@ -58,10 +59,18 @@ else if(type.baseType() == BaseContentType.WIDGET  && !Config.getBooleanProperty
       throw new ResourceNotFoundException("cannnot find contentlet id " +  argument + " lang:" + params.language);
     }
 
+    // If the time machine date is after the Publish date of the Contentlet (identifier data)
+    // we need to show the working. If not we only show the live.
+
+    boolean showWorking = false;
+
+    if (context.get("_show_working_") != null && (boolean)context.get("_show_working_")) {
+      showWorking = true;
+    }
 
 
 
-      return  File.separator +  params.mode.name() + File.separator   + argument + "_" + cv.getLang() + "." + EXTENSION;
+      return  File.separator + ( (!showWorking) ? params.mode.name() : PageMode.PREVIEW_MODE.name() ) + File.separator   + argument + "_" + cv.getLang() + "." + EXTENSION;
```

Comment:
1) You can use StringBuilder here
2) replace "_" by StringPool.UNDERLINE
3) replace "." by  StringPool.PERIOD
4) Agree parentheses are useless for showWorking

Reply:
done